### PR TITLE
Move build system to Maven

### LINKED ISSUE #337: Move build system to Maven
We want to create new builds of Scene Builder for every major release of the JavaFX SDK and the Java SDK. That means today, we want to build with and bundle with Java 16.
However, this is not yet supported by Gradle (see https://github.com/gradle/gradle/issues/13481) . Instead of going through issues with every release, we should use maven to build Scene Builder.

### PR DESCRIPTION
**Pull request by abhinayagarwal**
This PR removes Scene Builder's build system from Gradle to Maven

### Issue

Fixes #337

### Progress

- [x] Change must not contain extraneous whitespace
- [x] License header year is updated, if required
- [x] Verify the contributor has signed [Gluon Individual Contributor License Agreement (CLA)](https://docs.google.com/forms/d/16aoFTmzs8lZTfiyrEm8YgMqMYaGQl0J8wA0VJE2LCCY)

### CODE REVIEW

File: .github/scripts/jpackage.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+JAVAFX_MODULES=javafx.fxml,javafx.media,javafx.swing,javafx.web
+
+$JAVA_HOME/bin/jlink \
+--module-path $JAVAFX_HOME \
+--add-modules $jdeps_modules,$JAVAFX_MODULES \
+--output app/target/runtime \
+--strip-debug --compress 2 --no-header-files --no-man-pages
+
+$JPACKAGE_HOME/bin/jpackage \
+--app-version $VERSION \
```
Comment (jperedadnr) [inline .github/scripts/jpackage.sh:11]: Testing the script, seems `16.0.0-SNAPSHOT` is not accepted as a valid version.

File: .github/scripts/jpackage.sh [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+--input $GITHUB_WORKSPACE/app/target/lib \
+--install-dir /opt \
```
Comment (jperedadnr) [inline .github/scripts/jpackage.sh:13]: This should be `Applications` on Mac? With `/opt` the installer shows: 
```
Modifying “opt” requires an administrator name and password.
```

File: .github/scripts/jpackage.sh [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+--license-file LICENSE.txt \
+--main-jar SceneBuilder-$VERSION-all.jar \
+--main-class $MAIN_CLASS \
+--name SceneBuilder \
+--description "Scene Builder" \
+--vendor Gluon \
+--verbose \
+--runtime-image $GITHUB_WORKSPACE/app/target/runtime \
+--dest $INSTALL_DIR \
+"$@"
```
Comment (jperedadnr) [inline .github/scripts/jpackage.sh:23]: The installer fails with this exception:
```
/opt/SceneBuilder.app/Contents/MacOS/SceneBuilder
Graphics Device initialization failed for :  es2, sw
Error initializing QuantumRenderer: no suitable pipeline found
java.lang.RuntimeException: java.lang.RuntimeException: Error initializing QuantumRenderer: no suitable pipeline found
```

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+
+    <modules>
+        <module>kit</module>
+        <module>app</module>
+    </modules>
+
+    <properties>
+        <javafx.version>16</javafx.version>
+        <aether.version>1.1.0</aether.version>
+        <charm.glisten.version>5.0.0-jdk9</charm.glisten.version>
```
[the remark this answers is not in the document]
Comment (abhinayagarwal) [reply pom.xml:21]: Yes, better to do in a follow-up PR. I have logged an issue for it: https://github.com/gluonhq/scenebuilder/issues/339

File: README.md [hunk 5]
```diff
@@ -41,35 +42,32 @@ These are the requisites:
 
 To build the Scene Builder services, on the project's root, run:
 
-`./gradlew clean build`
+`mvn clean package`
 
-It will create a partial shadow cross-platform jar under `app/build/libs/scenebuilder-$version.jar`, that doesn't include the JavaFX dependencies.
+It will create a partial shadow cross-platform jar under `app/target/lib/scenebuilder-$version.jar`, that doesn't include the JavaFX dependencies.
```
[the remark this answers is not in the document]
Comment (abhinayagarwal) [reply README.md:47]: I am inclined to all lowercase :)

File: .github/scripts/jpackage.sh (hunk 1 above)
Comment (abhinayagarwal) [reply .github/scripts/jpackage.sh:11]: Yes, `VERSION` will never contain `-SNAPSHOT`

File: README.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (jperedadnr) [reply README.md:47]: That's okay, sure, as long as it is consistent.

**Review by erwin1 — APPROVED** [review]

**Review by jperedadnr — APPROVED** [review]

File: .github/workflows/win.yml [hunk 6]
```diff
@@ -29,13 +28,25 @@ jobs:
...
+          call mvn versions:set -DnewVersion=${{ env.VERSION }} -DgenerateBackupPoms=false
+          call mvn package -Dmaven.test.skip=true
+        env:
+          VERSION: ${{ steps.variables.outputs.SOURCE_VERSION }}
+      - name: Create Bundle using JPackage
+        shell: cmd
+        run: |
+          call .github\scripts\jpackage.bat
+          call ren app\target\SceneBuilder-${{ env.VERSION }}.msi SceneBuilder-${{ env.TAG }}.msi
+          call dir app\target\
```
Comment (tiainen) [inline .github/workflows/win.yml:47]: Be consistent with mac/linux and also use `$INSTALL_DIR`?

**Review by tiainen — APPROVED** [review]

**Pull request merged**